File: Group_Anagrams.py
# Link: https://leetcode.com/problems/group-anagrams/

class Solution(object):
    def groupAnagrams(self, strs):
        anagram = defaultdict(list)

        for word in strs:
            sorted_word = "".join(sorted(word))
            anagram[sorted_word].append(word)

        return list(anagram.values())

File: group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>>mp;
        for(string s: strs){
            string x = s;
            sort(s.begin(), s.end());
            mp[s].push_back(x);
        }
        vector<vector<string>>res;
        for(const auto &p : mp){
            res.push_back(p.second);
        }
        return res;
    }
};